sympy/core/function.py

```diff
@@ -944,93 +953,45 @@ class Derivative(Expr):
     This class also allows derivatives wrt non-Symbols that have _diff_wrt
     set to True, such as Function and Derivative. When a derivative wrt a non-
     Symbol is attempted, the non-Symbol is temporarily converted to a Symbol
-    while the differentiation is performed.
-
-    Note that this may seem strange, that Derivative allows things like
```

Comment:
I think this text should be kept, except with just talking about g(x) instead of cos(x)